### ISSUE TITLE
`--follow` and `--recursive` add seem to conflict

### ISSUE THREAD

**Issue by Uroc327** [twpayne/chezmoi#3772]
## Describe the bug

Issuing `chezmoi add --follow --recursive abc` (the `--recursive` is there for redundancy) when `.abc` is a symlink pointing to a directory, only the directory but not it's contents are added the chezmoi.

## To reproduce

```bash
mkdir -p test
touch test/a
touch test/b
touch test/c
ln -s test abc
chezmoi add --follow --recursive abc
chezmoi managed
# observe that abc/a, abc/b, abc/c, are missing and only abc is managed
```

## Expected behavior

All files `a`, `b`, and `c` should be managed.

**Comment by twpayne** [comment]
As a quick update on this, I did some investigation in [this branch](https://github.com/twpayne/chezmoi/compare/fix-3772). This is definitely a bug in chezmoi, but due to chezmoi's current implementation it's really hard to fix. chezmoi `stat`s the symlink (`abc` in the example you give) in multiple different places and needs to treat it as a directory for `--follow` but as a path component for `--recursive`.

**Comment by Direwolfesp** [comment]
What would be the workaround for this then? 

I have a dir `~/.config/hypr` linked to `~/dotfiles/hypr` and I wanna track the hypr directory but I dont care anymore about the `dotfiles/` dir, so if I apply my config from other pc I would like the `hypr/` to go to `~/.config/`.

Basically if I do `chezmoi add ~/dotfiles/hypr` and them clone it from other machine, i dont think they will be put in `~/.config/hypr`

**Comment by jdevera** [comment]
I hit this while migrating to chezmoi from my custom symlink driven script and I tried two workarounds that worked for me:

1. Hardly a workaround, but for one of those symlinked dirs, I added the contents one by one
2. For a second dir, I first replaced the symlink with a deep copy of the directory they were pointing to, and then added that recursively. I was okay with this since this is an infrequent operation. Something that helped check accuracy with this, was to temporarily rename the tracked dir in the dotfiles repo to have an `exact_` prefix because then I could use `chezmoi diff` to see exactly what was tracked and what I was missing, but be very careful with running `apply` since it will remove unmanaged files.

Overall, I think this bug is survivable, especially since it appears to be hard to crack, but it really needs more documentation about it.

**Comment by twpayne** [comment]
Agreed. I think the conflict of "follow this symlink in some cases, but not in others" is not easy to resolve in a user-friendly way.

#4695 adds an explicit error for this, which should help people in the future.
